core/queryalgebra/evaluation/src/main/java/org/eclipse/rdf4j/query/algebra/evaluation/impl/StrictEvaluationStrategy.java

```diff
@@ -1823,15 +1824,158 @@ public Value evaluate(ValueExprTripleRef node, BindingSet bindings)
 	 * @param ref      to evaluate
 	 * @param bindings with the solutions
 	 * @return iteration over the solutions
 	 */
 	public CloseableIteration<BindingSet, QueryEvaluationException> evaluate(TripleRef ref, BindingSet bindings) {
-		// not supported
-		return new EmptyIteration<BindingSet, QueryEvaluationException>();
+		// Naive implementation that walks over all statements matching (x rdf:type rdf:Statement)
+		// and filter those that do not match the bindings for subject, predicate and object vars (if bound)
+		final org.eclipse.rdf4j.query.algebra.Var subjVar = ref.getSubjectVar();
+		final org.eclipse.rdf4j.query.algebra.Var predVar = ref.getPredicateVar();
+		final org.eclipse.rdf4j.query.algebra.Var objVar = ref.getObjectVar();
+		final org.eclipse.rdf4j.query.algebra.Var extVar = ref.getExprVar();
+
+		final Value subjValue = getVarValue(subjVar, bindings);
+		final Value predValue = getVarValue(predVar, bindings);
+		final Value objValue = getVarValue(objVar, bindings);
+		final Value extValue = getVarValue(extVar, bindings);
+
+		// case1: when we have a binding for extVar we use it in the reified nodes lookup
+		// case2: in which we have unbound extVar
+		// in both cases:
+		// 1. iterate over all statements matching ((* | extValue), rdf:type, rdf:Statement)
+		// 2. construct a look ahead iteration and filter these solutions that do not match the
+		// bindings for the subject, predicate and object vars (if these are bound)
+		// return set of solution where the values of the statements (extVar, rdf:subject/predicate/object, value)
+		// are bound to the variables of the respective TripleRef variables for subject, predicate, object
+		// NOTE: if the tripleSource is extended to allow for lookup over asserted Triple values in the underlying sail
+		// the evaluation of the TripleRef should be suitably forwarded down the sail and filter/construct
+		// the correct solution out of the results of that call
+		if (extValue != null && !(extValue instanceof Resource)) {
+			return new EmptyIteration<>();
+		}
+
+		// whether the TripleSouce support access to RDF star
+		final boolean bSourceSupportsRdfStar = tripleSource instanceof RDFStarTripleSource;
+
+		// in case the
+		if (bSourceSupportsRdfStar) {
+			final CloseableIteration<? extends Resource, QueryEvaluationException> iter = ((RDFStarTripleSource) tripleSource)
+					.getRdfStarTriples((Resource) subjValue, (IRI) predValue, objValue);
+			return new LookAheadIteration<BindingSet, QueryEvaluationException>() {
+				@Override
+				protected BindingSet getNextElement()
+						throws QueryEvaluationException {
+					while (iter.hasNext()) {
+						Triple match = (Triple) iter.next();
+						if (subjValue != null && !subjValue.equals(match.getSubject())) {
+							continue;
+						}
+						if (predValue != null && !predValue.equals(match.getPredicate())) {
+							continue;
+						}
+						if (objValue != null && !objValue.equals(match.getObject())) {
+							continue;
+						}
```

Comment:
Why are these checks necessary? Isn't this check already supposed to be done by `RDFStarTripleSource.getRdfStarTriples`?

Reply:
It is like that to cover the case in which the implementation does not provide results that match the arguments but dump everything that is a `Triple` in the result iteration. Just a double check which does not put much overhead but ensure a correctness of the results.